File: ArrayCancellation.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    long long tests, n, coins;
    
    cin >> tests;
    
    while(tests--){
        coins = 0;
        cin >> n;
        vector<int>v1(n);

        for(int i = 0; i < n; i++){
            cin >> v1[i];
        }

        for(int i = 0; i < v1.size(); i++){
            for(int j = i+1; j < v1.size()-1; j++){
                if(v1[j] < 0 && v1[i] > 0){
                    int n = min(abs(v1[i]), abs(v1[j]));
                    v1[i]-=n;
                    v1[j]+=n;
                }
            }
        }

        for(int i = 0; i < v1.size(); i++){
            for(int j = i+1; j < v1.size(); j++){
                if(v1[i] < 0 && v1[j] > 0){
                    int n = min(abs(v1[i]), abs(v1[j]));
                    v1[j]-=n;
                    v1[i]+=n;
                    coins+=n;
                }    
            }
        }
       
        cout << coins << endl;
    }
}

File: ArrayCancellation.java
import java.math.BigInteger;
import java.util.Scanner;
import java.util.stream.Stream;

public class ArrayCancellation {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tests = sc.nextInt();

        for(int t = 0; t < tests; t++){
            int numbers = sc.nextInt();
            BigInteger coins;
            int[] arr = new int[numbers];

            for(int k = 0; k < numbers; k++){
                arr[k] = sc.nextInt();
            }

            doAllFreeOperations(arr);
            coins = doAllCostingOperations(arr);

            System.out.println(coins.toString());
        }

    }
    static void doAllFreeOperations(int[] arr){
        for(int i = 0; i < arr.length; i++){
            for(int j = i+1; j < arr.length-1; j++){
                while(arr[i] > 0 && arr[j] < 0){
                    arr[i]--;
                    arr[j]++;
                }
                    
            }
        }
    }
    static BigInteger doAllCostingOperations(int[] arr){
        BigInteger coins = BigInteger.ZERO;
        for(int j = 0; j < arr.length; j++){
            for(int i = j+1; i < arr.length; i++){
                while(arr[j] < 0 && arr[i] > 0){
                    arr[i]--;
                    arr[j]++;
                    coins = coins.add(BigInteger.ONE);
                }
                    
            }
        }
        return coins;
    }
}
